### ISSUE TITLE
Proposal: no-negated-condition

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#3740]
I'd like to forbid:
```js
if (!a) {
    console.log(1);
} else {
    console.log(2);
}
```
in favor of:
```js
if (a) {
    console.log(2);
} else {
    console.log(1);
}
```
JSCS has a rule of that effect named [disallowNotOperatorsInConditionals](http://jscs.info/rule/disallowNotOperatorsInConditionals).

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by platinumazure** [comment]
Only with an `else` present? Or always?

**Comment by BYK** [comment]
I'm guessing this rule wouldn't make sense when there's no `else`. I'm :+1: for adding this.

**Comment by mzgol** [comment]
My personal preference is it should be disallowed only with `else` present or in ternary operations but it might have an additional option to enable it for `else`-less `if` blocks as well.

**Comment by nzakas** [comment]
@mzgol would you like to try implementing this?

**Comment by mzgol** [comment]
@nzakas I'm afraid I have limited time within the next 2-3 weeks but I can do it later if no one gets to it before me.

**Comment by nzakas** [comment]
We also don't have the time to do new rules, so this will remain open until a volunteer steps forward.

**Comment by mzgol** [comment]
@nzakas Sure, no worries. :)

**Comment by michaelficarra** [comment]
For anyone interested, this would be an extremely easy first contribution. Just trigger on IfStatement and report if the test is a `UnaryExpression` with operator `!` and the alternate is `null`.
